### ISSUE TITLE
Upgrade of Sabre image uploading from Thunderbird not working

### ISSUE THREAD

**Issue by LEDfan** [owncloud/contacts#878]
@PVince81 the synchronising of images doesn't work after upgrading Sabre. I'm getting this error:
````
{"reqId":"aEDWwc1tYoTL5y7SGmhr","remoteAddr":"33.33.33.1","app":"PHP","message":"imagecreatefromstring(): Empty string or invalid image at \/var\/www\/owncloud\/lib\/private\/image.php#624","level":0,"time":"2015-04-11T12:36:42+00:00","method":"GET","url":"\/index.php\/apps\/contacts\/addressbook\/local\/1\/contacts"}
{"reqId":"aEDWwc1tYoTL5y7SGmhr","remoteAddr":"33.33.33.1","app":"core","message":"OC_Image->loadFromBase64, could not load","level":0,"time":"2015-04-11T12:36:43+00:00","method":"GET","url":"\/index.php\/apps\/contacts\/addressbook\/local\/1\/contacts"}
{"reqId":"G29Bo+6t1+VBc4VbsyN1","remoteAddr":"33.33.33.1","app":"no app in context","message":"Accessing the API class is deprecated! Use the appropriate services instead!","level":0,"time":"2015-04-11T12:36:43+00:00","method":"POST","url":"\/index.php\/apps\/contacts\/preference\/set"}
````

I had a look at it, but I don't know sabredav carddav well. 

There are many users who will have the same problem

**Comment by PVince81** [comment]
I don't know sabredav carddav either... but I can give it a try.

**Comment by DeepDiver1975** [comment]
Thanks for the hint @LEDfan - so sabredav changed it's behavior some point in time to either base64 encode binary data or not?

**Comment by LEDfan** [comment]
@DeepDiver1975 I had a look at it some time ago. I then found that the base64 string passed to the \OCP\Image::* function wasn't a valid image, so it couldn't be transformed. As a result the contacts app just shows the placeholder image. Back then I thought there was a problem (maybe encoding?) between the OC Database (where the images are stored) and sabredav. But I couldn't find the real cause.

**Comment by DeepDiver1975** [comment]
@LEDfan please verify this is fixed with #942 - THX

**Comment by LEDfan** [comment]
@DeepDiver1975 fixed! Great that you removed the small base64 encoded pictures!
